src/metabase/driver/sql/query_processor.clj

```diff
@@ -754,7 +754,21 @@
       (throw (ex-info "Summing intervals is not supported" {:args args})))
     (apply hx/call :+ (map (partial ->honeysql driver) args))))
 
-(defmethod ->honeysql [:sql :-] [driver [_ & args]] (apply hx/call :- (map (partial ->honeysql driver) args)))
+(defmethod ->honeysql [:sql :-]
+  [driver [_ & [first-arg & other-args :as args]]]
+  (assert (not (interval? first-arg))
```

Comment:
Is it ok to use "just assert"? Or should I prefer `(throw (ex-info (tru "..." {:type ...}))`?

Reply:
Broadly speaking, it's fine to use `assert`.

More specifically, since `assert` throws `AssertionError` (which is an `Error` not an `Exception`, and so isn't caught by `(catch Exception e ...)`) it should not be used for an error a user might have caused, only an error caused by bad coding.

Put differently, `assert` is for 500 errors, `(when ... (throw (ex-info ...)))` is for 4xx errors.